### PR TITLE
chore: implement launch command

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

File: packages/browsers/src/CLI.ts [hunk 1]
```diff
@@ -30,6 +31,16 @@ type Arguments = {
   platform?: BrowserPlatform;
 };
 
+type LaunchArgs = {
```
Comment (Lightning00Blade) [inline packages/browsers/src/CLI.ts:34]: How about we extract into `CommonArgs` then 
```
type LaunchArgs = CommonArgs & {
  detached: boolean;
 }

File: packages/browsers/src/launcher.ts [hunk 2]
```diff
@@ -23,8 +24,11 @@ import {
   executablePathByBrowser,
 } from './browsers/browsers.js';
 import {CacheStructure} from './CacheStructure.js';
+import {debug} from './debug.js';
 import {detectBrowserPlatform} from './detectPlatform.js';
 
+const debugLaunch = debug('puppeteer:browsers:fetcher');
```
Comment (Lightning00Blade) [inline packages/browsers/src/launcher.ts]:
```suggestion
const debugLaunch = debug('puppeteer:browsers:launcher');
```

File: packages/browsers/src/launcher.ts [hunk 3]
```diff
@@ -67,3 +71,214 @@ export function computeExecutablePath(options: Options): string {
...
+  kill(): void {
+    // If the process failed to launch (for example if the browser executable path
+    // is invalid), then the process does not get a pid assigned. A call to
+    // `proc.kill` would error, as the `pid` to-be-killed can not be found.
+    if (
+      this.#browserProcess &&
+      this.#browserProcess.pid &&
+      pidExists(this.#browserProcess.pid)
+    ) {
+      const proc = this.#browserProcess;
```
Comment (Lightning00Blade) [inline packages/browsers/src/launcher.ts]: Why reassign here? Imo it makes it harder to read.

File: packages/browsers/src/launcher.ts [hunk 4]
```diff
@@ -67,3 +71,214 @@ export function computeExecutablePath(options: Options): string {
...
+
+    opts.pipe ??= false;
+    opts.dumpio ??= false;
+    opts.handleSIGINT ??= true;
+    opts.handleSIGTERM ??= true;
+    opts.handleSIGHUP ??= true;
+    opts.detached ??= true;
+
+    let stdio: Array<'ignore' | 'pipe'>;
+    if (opts.pipe) {
```
Comment (Lightning00Blade) [inline packages/browsers/src/launcher.ts]: Nit: Maybe extract into method

**Review by Lightning00Blade — APPROVED** [review]
LGTM + some minor comments

File: packages/browsers/src/CLI.ts (hunk 1 above)
Comment (OrKoN) [reply packages/browsers/src/CLI.ts:34]: The reason for separating it is that it is easy to add things to common args and forget to apply it in all commands so things will grow and be unused.

File: packages/browsers/src/CLI.ts [hunk 5]
```diff
@@ -79,6 +90,51 @@ export class CLI {
...
         defaultDescription: 'Auto-detected by default.',
       })
+      .command(
+        'launch <browser>',
+        'Launch the specified browser',
+        yargs => {
+          yargs.positional('browser', {
+            description: 'The browser version',
+            type: 'string',
+            coerce: (opt): InstallArgs['browser'] => {
```
Comment (Lightning00Blade) [inline packages/browsers/src/CLI.ts]: Nit
```suggestion
            coerce: (opt): LaunchArgs['browser'] => {
```

File: packages/browsers/src/launcher.ts (hunk 3 above)
Comment (OrKoN) [reply packages/browsers/src/launcher.ts]: for a shorter variable name?

File: packages/browsers/src/CLI.ts (hunk 1 above)
Comment (Lightning00Blade) [reply packages/browsers/src/CLI.ts:34]: ACK

File: packages/browsers/src/launcher.ts (hunk 3 above)
Comment (Lightning00Blade) [reply packages/browsers/src/launcher.ts]: We have `proccess` and `proc` not very easy to distinguish.

File: packages/browsers/src/launcher.ts (hunk 4 above)
Comment (OrKoN) [reply packages/browsers/src/launcher.ts]: Extracted.

File: packages/browsers/src/launcher.ts (hunk 3 above)
Comment (OrKoN) [reply packages/browsers/src/launcher.ts]: Thanks, renamed to use the attribute directly.

**Comment by Hakiking** [comment]
[CI](https://github.com/puppeteer/puppeteer/actions/workflows/ci.yml)

**Comment by Hakiking** [comment]
<details><summary>Details</summary>
<p>



</p>
</details>

**Pull request merged**